### PR TITLE
[TypeDeclaration] Do not remove multiple docblocks with comment on TypedPropertyFromAssignsRector

### LINKED ISSUE #7693: TypedPropertyFromAssignsRector seems to delete too many comments
# Bug Report

Hi, it seems that TypedPropertyFromAssignsRector rule delete too many comments above the property when it's applied. I guess it's due to the VarTagRemover involved in the rule ? 

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-main      |
|0.15.1  | 7.4  |

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/88c48c30-05e1-4e29-b5b5-22fe0339e324

```php
<?php

final class Demo
{
    // A comment
    
    /**
    * Another comment
    */
    
    /**
     * @var \DateTime
     */
    private $property;

    public function __construct()
    {
        $this->property = new \DateTime();
    }
}
```

### Responsible rules

* `TypedPropertyFromAssignsRector`

## Actual behavior

```diff
 final class Demo
 {
-    // A comment
-    
-    /**
-    * Another comment
-    */
-    
-    /**
-     * @var \DateTime
-     */
-    private $property;
+    private \DateTime $property;
 
     public function __construct()
     {
```

## Expected Behavior

```diff
 final class Demo
 {
    // A comment
    
    /**
    * Another comment
    */

-    /**
-     * @var \DateTime
-     */
-    private $property;
+    private \DateTime $property;
 
     public function __construct()
     {
​```

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/7693

Multiiple docblock should only remove last doc as detected one is the last:

```diff
    // A comment

    /**
     * Another comment
     */

-    /**
-     * @var \DateTime
-     */
```

For only `comment + docblock`, it remove the docblock part:

```diff
   // A comment
-    /**
-     * @var \DateTime
-     */
```

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed :tada: /cc @neveldo

**Comment by samsonasik** [comment]
All checks have passed :tada: @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
The safest way seems mark previous docs as comment.

**Comment by samsonasik** [comment]
I  updated to mark Doc as Comment when not in last, as the used one is last https://github.com/rectorphp/rector-src/pull/3263/commits/06b3f6e7bb6521771916babaeed15099c04a8e0a

File: src/Exception/ShouldNotHappenException.php [hunk 1]
```diff
@@ -33,7 +33,6 @@ private function createDefaultMessageWithLocation(): string
         $method = $class !== null ? ($class . '::' . $function) : $function;
 
         /** @var string $method */
-        /** @var int $line */
```
Comment (samsonasik) [inline src/Exception/ShouldNotHappenException.php]: mark previous docs as comment seems make this `@var int $line` removed.

Comment (samsonasik) [reply src/Exception/ShouldNotHappenException.php]: Resolved with apply setDocComment after set comments attributes https://github.com/rectorphp/rector-src/pull/3263/commits/2413173a664c15e79d5c99d36105ead0d4c12eb3

**Comment by samsonasik** [comment]
Next todo: preserve spacing  new line between docblocks/comments when calling `PhpDocInfoFactory->createFromNode()`.

**Comment by TomasVotruba** [comment]
I see. Is this ready for review?

**Comment by samsonasik** [comment]
Yes, but not mergeable yet, as currently just calling:

```
PhpDocInfoFactory->createFromNode()
```

will remove the spacing, that will cause many unnecessary space removal for something like:.

```
/**
 * @license abc
 */

/**
 * @Annotation
 */
```

I can split into smaller PR for change this part only (comment + single docblock):

```diff
   // A comment
-    /**
-     * @var \DateTime
-     */
```
 for non-multiple docblocks, wdyt?

**Comment by samsonasik** [comment]
I added failing fixture to preserve multiple docs when docs never changed https://github.com/rectorphp/rector-src/pull/3263/commits/04929fb2e9d42729d6d749ac2a189edae62a2c72 , it currently still removing space new line even docs never changed, which invalid.

```diff
 final class PreserveMultipleDocsNoChange
 {
     // A comment
-
     /**
      * Another comment
      */
-
     /**
      * @var class-string
      */
-    private $property;
+    private string $property;
```

**Comment by samsonasik** [comment]
@TomasVotruba it finally resolved now 🎉 🎉  🎉  , on after phpdocinfo changed, when no longer children, it needs to apply

- mark previous comments + docs as comments
- renew last doc as new main doc

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Review by TomasVotruba — APPROVED** [review]

**Pull request merged**